### PR TITLE
Add default parameters for xboost

### PR DESCRIPTION
**Pull request by yuhong-sigopt**

### CODE REVIEW

File: sigopt/xgboost/utils.py [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+  'reg:squaredlogerror': 'rmsle',
+  'reg:logistic': 'rmse',
+  'reg:pseudohubererror': 'mphe',
+  'binary:logistic': 'logloss',
+  'binary:logitraw': 'logloss', # >= 1.4.0
+  'binary:hinge': 'error',
+  'count:poisson': "poisson-nloglik",
+  'survival:cox': "cox-nloglik",
+  'survival:aft': 'aft-nloglik',
+}
```
[the remark this answers is not in the document]
Comment (yuhong-sigopt) [reply sigopt/xgboost/utils.py:92]: These code is wrote before I think about use `bst.save_config()`. I keep these code as a library when we need to know the defaults values of XGBoost when we do not have a trained booster.

File: sigopt/xgboost/utils.py [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+
+
+def get_objective_default_eval_metric(objective, xgb_version=None):
```
[the remark this answers is not in the document]
Comment (yuhong-sigopt) [reply sigopt/xgboost/utils.py:95]: As same above, it is not used. In the latest submit, metrics is got as the way you mentioned.

File: sigopt/xgboost/utils.py [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+  train_default_args = get_default_args(xgb.train)
+  defaults = {k: train_default_args[k] for k in train_parameters}
+
+  if params is not None:
+    defaults['booster'] = 'gbtree'
+    if include_all:
+      defaults.update(dart_booster_defaults)
+      defaults.update(linear_booster_defaults)
+      defaults.update(tree_booster_defaults)
+      defaults.update(tweedie_regression_defaults)
```
[the remark this answers is not in the document]
Comment (yuhong-sigopt) [reply sigopt/xgboost/utils.py:119]: Same as above, it is not used.

File: sigopt/xgboost/utils.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (harvey-intel) [reply sigopt/xgboost/utils.py:92]: ok. if they are not being used, then we should not leave them here. We would always have access to the booster object in the integration setting.

**Comment by yuhong-sigopt** [comment]
Example: 
  https://github.com/ericlee0803/xgb-integration-testing/blob/d33139a84d8b01da877de0ed324741aef1dcac0d/model.py

Usage:
 python model.py

Run example: 
   https://app.sigopt.com/run/100194

Screenshot:
image

File: sigopt/xgboost/utils.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (yuhong-sigopt) [reply sigopt/xgboost/utils.py]: Not used code is removed.

File: sigopt/run_context.py [hunk 4]
```diff
@@ -357,6 +371,10 @@ def _create_checkpoint(self, body):
   def _set_parameters(self, parameters):
     return parameters
 
+  @updates('all_assignments')
```
Comment (taylorjacklespriggs) [inline sigopt/run_context.py:374]: is there a spec for an API change to support this?

File: sigopt/xgboost/utils.py [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+  wrapper_specific = {
+    'importance_type', 'kwargs', 'missing', 'n_estimators', 'use_label_encoder',
+    'enable_categorical'
+  }
+
+  filtered = {}
+  for k, v in params.items():
+    if k not in wrapper_specific and not callable(v):
+      filtered[k] = v
+  return filtered
```
Comment (harvey-intel) [inline sigopt/xgboost/utils.py:68]: What are these for?

Comment (yuhong-sigopt) [reply sigopt/xgboost/utils.py:68]: I will remove the filter. I copy the code from ski-learn.

File: sigopt/xgboost/utils.py [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+      continue
+  return str(value)
+
+def parse_metrics(value):
+  if isinstance(value, str):
+    return value
+  if isinstance(value, list):
+    if len(value) == 1:
+      return value[0]
+  return str(value)
```
[the remark this answers is not in the document]
Comment (yuhong-sigopt) [reply sigopt/xgboost/utils.py:34]: I prefer to keep it, as user may use define objective function.

[the remark this answers is not in the document]
Comment (yuhong-sigopt) [reply sigopt/xgboost/utils.py:34]: I see, will remove it.

**Comment by harvey-intel** [comment]
Yuhong, let's leave this PR open for now until we decide on how to design the AssignmentGroup on the API side.

**Comment by yuhong-sigopt** [comment]
> Yuhong, let's leave this PR open for now until we decide on how to design the AssignmentGroup on the API side.

ok.

**Pull request merged**